src/onchain/swap.rs

```diff
@@ -199,10 +199,19 @@ impl Swap {
         sats_per_vbyte: u32,
         lsp_fee_param: Option<OpeningFeeParams>,
     ) -> Result<String> {
-        let swap_address_info = self.create(lsp_fee_param.clone()).map_to_runtime_error(
-            RuntimeErrorCode::NodeUnavailable,
-            "Couldn't generate swap address",
-        )?;
+        let lsp_fee_param = lsp_fee_param.unwrap_or(
+            self.support
+                .calculate_lsp_fee_for_amount(failed_swap_info.amount.sats, Some(TWO_WEEKS))?
+                .lsp_fee_params
+                .ok_or_permanent_failure("lsp_fee_param is expected to never be None")?, // todo stop treating lsp_fee_param as an Option anywhere. In reality it is always Some(). Leaving it as is for now, to not break backwards-compatibility
+        );
+
+        let swap_address_info = self
+            .create(Some(lsp_fee_param.clone()))
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Couldn't generate swap address",
+            )?;
```

Comment:
Right. But there is a problem in `ChannelCloses::swap()`. We don't have an amount there and `calculate_lsp_fee_for_amount` requires one, in order to be able to return the calculated `lsp_fee`. Ironically, we don't even need the nominal `lsp_fee`, we're just interested in the `lsp_fee_params`.

So with https://github.com/getlipa/lipa-lightning-lib/pull/1277/commits/5f986a8626cc0cb2723db246768cf9d40071403d I changed the function such that the payment amaount parameter is optional and consequently, the returned `lsp_fee` also becomes optional (it can only be calculated when the payment amount is passed in to the function). So this is quite some change, checkout the UDL file the get a quick overview.

A better solution still would be to have 2 functions to get rid of the `Option`s. But maybe this does it for now.